feat(releases): Add Archiving Releases section (#2738)
getsentry/sentry-docs: src/docs/product/releases/index.mdx

JavaScript and React Native projects can utilize release artifacts to unminify or symbolicate error stack traces, you can find the artifacts uploaded to Sentry by navigating to **[Project] » Project Settings » Source Maps**.

To learn more, please check out our [Source Maps](/platforms/javascript/sourcemaps/) docs.

## Archiving Releases

> If you don't wish to see a release anymore you can archive it either using CLI:

Reviewer: This is a great start, but can we say a bit more about the why a customer might benefit from archiving a release? Similar to https://sentry-docs-git-feat-release-archiving.sentry.dev/product/sentry-basics/environments/#hidden-environments?

```bash
sentry-cli releases archive VERSION
```

> or in the UI:

Reviewer: ```suggestion
or in sentry.io:
```

![archive release ui](archive-release-ui.png)

Archived releases won't show up in Releases view unless you filter for them:

![archived releases ui](archived-releases-ui.png)

> You can always restore archived release either in the UI or using CLI:

Reviewer: ```suggestion
You can restore archived release either in sentry.io or by using CLI:
```

```bash
sentry-cli releases restore VERSION

> ```

Reviewer: Can we add a screen shot of how to restore a release in sentry.io?

JavaScript and React Native projects can utilize release artifacts to unminify or symbolicate error stack traces, you can find the artifacts uploaded to Sentry by navigating to **[Project] » Project Settings » Source Maps**.

To learn more, please check out our [Source Maps](/platforms/javascript/sourcemaps/) docs.

## Archiving Releases

> Since the received event from the deleted release would automatically recreate it, Sentry has an option to hide releases permanently from the UI. Archiving releases can be useful when you're no longer interested in seeing them or you've accidentally created a release you no longer want.

Reviewer: ```suggestion
Since an event received from a deleted release automatically recreates it, Sentry has an option to hide releases permanently from sentry.io. In addition, archiving releases can be useful if you're no longer interested in viewing the release or if you've accidentally created a release you no longer want to track.
```
